### PR TITLE
Support yaml testbed file

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

The goal here is to support yaml testbed file with the support of nested
data structures that could be applicable for multi-DUT scenario.
1. Enable `testbed-cli.sh` the capacity to read from `yaml` testbed file.
2. Enable Ansible module `test_facts` with the capacity to read from `yaml` testbed file.
3. Enable `tbinfo` the support of yaml format testbed file and keep the csv
format compatible. Currently, `TestbedInfo` will prefer using `yaml`
testbed file if there exists one even `--testbed-file` passes a `csv`
testbed file. If `--testbed-file` passes a `csv` file and there is no
yaml testbed file, `TestbedInfo` will try to dump a yaml testbed file
for later use.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Support `yaml` testbed file and deprecate `csv` format in the future.

#### How did you do it?
Add support of reading `yaml` file to:
1. `testbed-cli.sh`
2. `test_facts` module
3. `TestbedInfo` class

#### How did you verify/test it?
1. run deploy minigraph with/without `yaml` testbed file present.
2. run testcase with/without `yaml` testbed file present.

### CODE REVIEW

**Comment by lolyu** [comment]
Retest this please

**Comment by yxieca** [comment]
@lolyu change looks good. But did you forget to include a new testbed.yaml? Speaking of which, can you provide a tool to convert existing testbed.csv to testbed.yaml? These changes can come as a separate PR.

**Comment by lolyu** [comment]
> @lolyu change looks good. But did you forget to include a new testbed.yaml? Speaking of which, can you provide a tool to convert existing testbed.csv to testbed.yaml? These changes can come as a separate PR.

Sure, I will create another PR to do this.

**Review by wangxin — APPROVED** [review]

**Comment by lolyu** [comment]
Hi, @wangxin, I've added another two commits which moves `TestbedInfo` class out of `conftest.py`, could you please take a review again, thanks!

**Review by yxieca — APPROVED** [review]

**Comment by yxieca** [comment]
@lolyu did you test without testbed.csv (only having testbed.yaml)?

**Comment by lolyu** [comment]
Hi, @yxieca, I've tested `bgp_facts` with `testbed.yaml` only and it looks good to me.

File: ansible/vtestbed.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  ptf_image_name: docker-ptf
+  ptf: ptf-03
+  ptf_ip: 10.250.0.108/24
+  ptf_ipv6: fec0::ffff:afa:8/64
+  server: server_1
+  vm_base: VM0104
+  dut:
+    - vlab-04
+  comment: Tests virtual switch vm
+- conf-name: vms-kvm-dual-t0
```
Comment (wangxin) [inline ansible/vtestbed.yaml:50]: Would it be more readable if insert a blank line between each test setup?

Comment (lolyu) [reply ansible/vtestbed.yaml:55]: Updated, please take a review!

**Pull request merged**